File: Core/Sources/Metadata/git.c
#ifndef GIT_TAG
#define GIT_TAG "unknown"
#endif

#ifndef GIT_COMMIT
#define GIT_COMMIT "unknown"
#endif

const char* git_tag_string(void) {
    return GIT_TAG;
}

const char* git_commit_string(void) {
    return GIT_COMMIT;
}

File: Core/Sources/Core/Metadata/Git.swift
import Metadata

public struct PackageMetadata {
    public static var gitTag: String? {
        let gitTag = String(validatingCString: git_tag_string())
        return if gitTag == "unknown" {
            nil
        } else {
            gitTag
        }
    }
    public static var gitCommit: String? {
        let gitCommit = String(validatingCString: git_commit_string())
        return if gitCommit == "unknown" {
            nil
        } else {
            gitCommit
        }
    }
}
